docs: add a documentation page detailing ESLint NgRx plugin and how to use it (#3208)
ngrx/platform: projects/ngrx.io/content/guide/eslint-plugin.md

# ESLint plugin for NgRx

## What is it?

Use [ESLint](https://eslint.org/) to follow the best practices and to avoid common pitfalls in your application.

The [NgRx ESLint Plugin](https://github.com/timdeschryver/eslint-plugin-ngrx) is no different and promotes the key concepts to create a maintainable project. It consists of @ngrx/store, @ngrx/effects, and @ngrx/component-store rules and a handful of preconfigured configurations.

> For more information, see the [documentation page](https://github.com/timdeschryver/eslint-plugin-ngrx/blob/main/README.md).

Reviewer: This link is already added in the next line, so I think we could remove this one.

```suggestion
```

Author: Also a good point

> The plugin comes with a number of rules that help address most popular NgRx malpractices. The rules are configurable so that every developer can choose which ones they necessarily want to follow, and which ones should givbe a linting error or warning, and so on. A detailed documentation of all the rules with examples can be found [here](https://github.com/timdeschryver/eslint-plugin-ngrx/tree/main/docs/rules). Some rules also allow automatic fixes with `ng lint --fix`.

Reviewer: ```suggestion
The plugin comes with a number of rules that help address most popular NgRx malpractices. The rules are configurable so that you can choose the ones you want to follow, and which rules should give a linting error or warning. 

A detailed documentation of all the rules with examples can be found on the [documentation page](https://github.com/timdeschryver/eslint-plugin-ngrx/blob/main/README.md#rules). 

Some rules also allow automatic fixes with `ng lint --fix`.
```

Author: Yeah, using "you" instead of "developer" sounds more friendly, I like it

## Installation

### Adding with NgRx

If you use `ng add @ngrx/store` to install NgRx itself, and you already use ESLint, the schematic automatically adds the plugin.

### Installing with `ng add`

If you already have @ngrx/store installed, you can add the plugin manually with the following command:

```sh
ng add eslint-plugin-ngrx
```

> This uses configures and enables the recommended rules.

Reviewer: ```suggestion
This installs the package and configures ESLint to enable the recommended rules.
```
